hosting/views.py

```diff
@@ -644,62 +644,86 @@ class OrdersHostingDetailView(LoginRequiredMixin,
     model = HostingOrder
 
     def get_object(self):
-        return HostingOrder.objects.filter(
+        return HostingOrder.objects.get(
             pk=self.kwargs.get('pk')) if self.kwargs.get('pk') else None
 
     def get_context_data(self, **kwargs):
         # Get context
         context = super(DetailView, self).get_context_data(**kwargs)
         obj = self.get_object()
         owner = self.request.user
-        if 'specs' not in self.request.session:
-            return HttpResponseRedirect(
-                reverse('hosting:create_virtual_machine'))
-        if 'token' not in self.request.session:
-            return HttpResponseRedirect(reverse('hosting:payment'))
         stripe_customer_id = self.request.session.get('customer')
         customer = StripeCustomer.objects.filter(id=stripe_customer_id).first()
         stripe_utils = StripeUtils()
-        card_details = stripe_utils.get_card_details(customer.stripe_id,
-                                                     self.request.session.get(
-                                                         'token'))
-        if not card_details.get('response_object'):
-            msg = card_details.get('error')
-            messages.add_message(self.request, messages.ERROR, msg,
-                                 extra_tags='failed_payment')
-            return HttpResponseRedirect(
-                reverse('hosting:payment') + '#payment_error')
+        if customer:
+            card_details = stripe_utils.get_card_details(
+                customer.stripe_id,
+                self.request.session.get('token')
+            )
+        else:
+            card_details = {}
 
-        if self.request.GET.get('page', '') == 'payment':
+        if self.request.GET.get('page') == 'payment':
             context['page_header_text'] = _('Confirm Order')
         else:
             context['page_header_text'] = _('Invoice')
 
         if obj is not None:
+            # invoice for previous order
             try:
-                manager = OpenNebulaManager(email=owner.email,
-                                            password=owner.password)
+                manager = OpenNebulaManager(
+                    email=owner.email, password=owner.password
+                )
                 vm = manager.get_vm(obj.vm_id)
                 context['vm'] = VirtualMachineSerializer(vm).data
             except WrongIdError:
-                messages.error(self.request,
-                               'The VM you are looking for is unavailable at the moment. \
-                                Please contact Data Center Light support.'
-                               )
+                messages.error(
+                    self.request,
+                    'The VM you are looking for is unavailable at the moment. \
```

Comment:
Not related to this PR, but translating texts shown to the user where possible would be great.